code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -200,6 +218,15 @@
 	use_power(power_usage)
 	update_parents()
 
+/obj/machinery/atmospherics/components/unary/thermomachine/attackby(obj/item/assembly/signaler/anomaly/pyro, mob/living/user, params)
+	if(!has_pyro_anomaly_core && allows_cooling)
+		to_chat(user, span_notice("The anomaly core fits perfectly!"))
+		has_pyro_anomaly_core = TRUE
+		qdel(pyro)
+		RefreshParts()
+	else
+		to_chat(user, span_notice("There is already a anomaly core slotted in."))
```

Comment:
```suggestion
		balloon_alert(user, "already an anomaly core!")
```

Also this should be an early return

Reply:
Changed code around a bit, let me know if it needs any other adjustments